pkg/operation/botanist/dnsrecord.go

```diff
@@ -206,3 +211,38 @@ func (b *Botanist) deployOrRestoreDNSRecord(ctx context.Context, dnsRecord compo
 	}
 	return dnsRecord.Deploy(ctx)
 }
+
+// CleanupOrphanedDNSRecordSecrets cleans up secrets related to DNSRecords which may be orphaned after introducing the 'dnsrecord-' prefix
+func (b *Botanist) CleanupOrphanedDNSRecordSecrets(ctx context.Context) error {
+	internalDNSRecordSecretName := b.Shoot.Components.Extensions.InternalDNSRecord.GetSecret().Name
+	ownerDNSRecordSecretName := b.Shoot.Components.Extensions.OwnerDNSRecord.GetSecret().Name
+
+	if strings.HasPrefix(internalDNSRecordSecretName, "dnsrecord-") && strings.HasPrefix(ownerDNSRecordSecretName, "dnsrecord-") {
+		err := b.cleanupOrphanedDNSRecordSecret(ctx, strings.TrimPrefix(internalDNSRecordSecretName, "dnsrecord-"))
+		if err != nil {
+			return err
+		}
+	}
+
+	externalDNSRecordSecretName := b.Shoot.Components.Extensions.ExternalDNSRecord.GetSecret().Name
+	ingressDNSRecordSecretName := b.Shoot.Components.Extensions.IngressDNSRecord.GetSecret().Name
+
+	if strings.HasPrefix(externalDNSRecordSecretName, "dnsrecord-") && strings.HasPrefix(ingressDNSRecordSecretName, "dnsrecord-") {
+		err := b.cleanupOrphanedDNSRecordSecret(ctx, strings.TrimPrefix(externalDNSRecordSecretName, "dnsrecord-"))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *Botanist) cleanupOrphanedDNSRecordSecret(ctx context.Context, secretName string) error {
+	err := b.K8sSeedClient.Client().Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: b.Shoot.SeedNamespace}})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
```

Comment:
```suggestion
	return client.IgnoreNotFound(b.K8sSeedClient.Client().Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: b.Shoot.SeedNamespace}}))
```
And since there is not much left of this method I would also remove the method itself and use the above one-liner directly.

Reply:
done.